### ISSUE TITLE
Span of Expr does not always cover the full expression.

### ISSUE THREAD

**Issue by GilShoshan94** [dtolnay/syn#1379]
Hi,

I have expressions in the arguments of my attribute macro (seperated by comma).
As such:
```rust
#[block(name = MyName, output = [aa, bb, cc], config = config, config)]
fn foo( in1: u32, config: &MyConfig) -> (Option<i32>, u32, u8) {
    todo!()
}
```

So in this example, I defined twice `config`, and it can either be just my keyword `config` alone, which will default to the ident of the same keyword, or followed by `= <ident>` to give a different name.

Here since I defined it twice, I want to report the error to the user and uderline them with the correct span.

This is how I save the span while parsing:
```rust
impl Parse for ConfigStruct {
    fn parse(input: ParseStream) -> Result<Self> {
        let span = match input.fork().parse::<syn::Expr>() {
            Ok(expr) => {
                let a = expr.to_token_stream();
                println!("{a}");
                // expr.span()
                a.span()
            },
            Err(_) => input.span(),
        };
        
        let _kw_token = input.parse()?;
        let lookahead = input.lookahead1();
        let (_eq_token, value) = if lookahead.peek(Token![=]) {
            (Some(input.parse()?), input.parse()?)
        } else {
            ...
```

As you can see, I tried to get the span on the `Expr` and also on the `TokenStream`.

Here is the output when I run my test:
![image](https://user-images.githubusercontent.com/29143941/218505156-1fbc8dac-a69f-4b67-9dce-cbe91472344f.png)

The print shows that indeed the expression is well parsed.
I expected the span to cover the same `TokenStream`, I expected the whole `config = config` to be underlined.

Am I missing something or is there an issue here ?

**Comment by dtolnay** [comment]
I guess you are on a stable compiler.

Spans on a stable compiler can only cover 1 token. See https://docs.rs/proc-macro2/1.0.51/proc_macro2/struct.Span.html#method.join.

**Comment by GilShoshan94** [comment]
Ah, ok.

I imagined it would work differently because it does work for an array `[arg, arg2, ...]`, the span here does cover correctly the full `syn::ExprArray`.

Checking into the code, I couldn't find a usage of `Span.join()` so I didn't think it was related.
I also saw that there is something special going on for `Group` and since the  `ExprArray` is group, I can see why it would work for that.

I will patiently wait for join() to come to the stable compiler than.

Thank you for taking the time to answer me.
Have a nice day.
